exchange/utils.go

```diff
@@ -202,6 +205,9 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 				privacy.ScrubDeviceIDsIPsUserDemoExt(reqWrapper, ipConf, "eids", false)
 			}
 		}
+		if buyerUIDSet && reqWrapper.User.BuyerUID == "" {
```

Comment:
I was thinking, could this be re-written, removing the lines 191-194, and instead just put the conditions here:

```
if reqWrapper.User != nil && reqWrapper.User.BuyerUID != "" && reqWrapper.User.BuyerUID == "" {
	rs.me.RecordAdapterBuyerUIDScrubbed(bidderRequest.BidderCoreName)
}
```
What do you think about this?

Reply:
Ah if you look at your suggested snippet again you'll see that you're referencing `reqWrapper.User.BuyerUID` twice. The reason I wrote it the way I did is so I can capture `reqWrapper.User.BuyerUID` as `buyerUIDSet` before it is potentially overwritten by the scrubber logic.
I am then checking if the value was set and has since been cleared by the scrubber before logging the metric.
Does that make sense?